synapse/storage/databases/main/monthly_active_users.py

```diff
@@ -355,16 +355,26 @@ async def populate_monthly_active_users(self, user_id):
             elif now - last_seen_timestamp > LAST_SEEN_GRANULARITY:
                 await self.upsert_monthly_active_user(user_id)
 
-    async def remove_deactivated_user(self, user_id: str) -> None:
+    async def remove_deactivated_user_from_mau_table(self, user_id: str) -> None:
         """
         Removes a deactivated user from the monthly active user
-        table and resets the "get_monthly_active_count" cache.
+        table and resets affected caches.
 
         Args:
             user_id(str): the user_id to remove
         """
 
-        await self.db_pool.simple_delete_one(
-            table="monthly_active_users", keyvalues={"user_id": user_id}
+        rows_deleted = await self.db_pool.simple_delete(
+            table="monthly_active_users",
+            keyvalues={"user_id": user_id},
+            desc="simple_delete",
         )
-        await self.invalidate_cache_and_stream("get_monthly_active_count", ())
+
+        if rows_deleted != 0:
+            await self.invalidate_cache_and_stream(
+                "user_last_seen_monthly_active", (user_id)
```

Comment:
ah, you've fallen into one of python's traps here. `(foo)` *looks* like a tuple, but is actually the same as `foo`. You need a trailing comma to force a tuple:

```suggestion
                "user_last_seen_monthly_active", (user_id,)
```